api/v1alpha1/database_webhook.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/nutanix-cloud-native/ndb-operator/api"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/common/util"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var databaselog = logf.Log.WithName("database-resource")
+
+func (r *Database) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-ndb-nutanix-com-v1alpha1-database,mutating=true,failurePolicy=fail,sideEffects=None,groups=ndb.nutanix.com,resources=databases,verbs=create;update,versions=v1alpha1,name=mdatabase.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Database{}
+
+func ndbSpecDefaulterForCreate(r *Database) {
+	if !r.Spec.NDB.SkipCertificateVerification {
+		r.Spec.NDB.SkipCertificateVerification = false
+	}
+
+}
+
+func instanceSpecDefaulterForCreate(r *Database) {
+
+	if r.Spec.Instance.DatabaseInstanceName == "" {
+		r.Spec.Instance.DatabaseInstanceName = api.DefaultDatabaseInstanceName
+	}
+
+	if len(r.Spec.Instance.DatabaseNames) == 0 {
+		r.Spec.Instance.DatabaseNames = api.DefaultDatabaseNames
```

Comment:
I see your point but I will hold this discussion for the another uber item that I have added in the task sheet. It is "Audit of the required fields and the default values". 

That itself won't be a big change but it will still need a discussion and some consensus.